src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -83,7 +99,7 @@ def get(self, request: Request, project) -> Response:
                 scope.set_tag("integration_provider", provider)
                 scope.set_tag("stacktrace_link.platform", platform)
 
-                if not filepath.startswith(config.stack_root):
+                if not filepath.startswith(config.stack_root) and not frame:
```

Comment:
following up from this comment https://github.com/getsentry/sentry/pull/35349#discussion_r890565081 - there is a difference between `stack_root_mismatch` and `file_not_found`. 

`stack_root_mismatch` means we weren't able to match up your integration configuration with the files in the stack trace, e.g. if the filepath `src/code.py` but the `stack_root` is `/bin` or something, we don't find a match

`file_not_found` would be like if the `stack_root` was `src/` and we made a request to the service (GH or whatever) and the file `code.py` wasn't found

Reply:
Ahh, makes sense. I added a test case to cover the scenario when the `file` does not match the `stack_root` as well as the `munged_filepath` does not match the `stack_root`. 

The logic is a bit confusing with all the branching but I think the tests should cover the scenarios.

Reply:
I feel yeah I wrote this stuff like 2 years ago, I'm sure it could be improved lol